### PR TITLE
[Serve] Feature flag and turn off placement group usage.

### PR DESCRIPTION
**Pull request by simon-mo**
## Related issue number
Due to https://github.com/ray-project/ray/issues/15801, Serve is having trouble upscaling with autoscaling cluster. Therefore we will turn off the use of placement for now.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/backend_state.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ class ReplicaState(Enum):
 
 
 ALL_REPLICA_STATES = list(ReplicaState)
+USE_PLACEMENT_GROUP = False
```
Comment (ericl) [inline python/ray/serve/backend_state.py:36]: Env var?

**Review by ericl — APPROVED** [review]
Rubber stamped

**Review by edoakes — APPROVED** [review]

**Comment by simon-mo** [comment]
I'm a bit stuck. @edoakes any clue why the `test_deploy` might time out (runtime doubled on my local machine). https://buildkite.com/ray-project/ray-builders-pr/builds/5451#2df15f8f-414f-4615-95e7-6ead1e782d02/6-11133

**Comment by edoakes** [comment]
Hmm there's nothing obvious to me, I would expect this would be strictly faster without placement groups...

**Comment by edoakes** [comment]
@simon-mo there were merge conflicts with my other PR. I fixed them to trigger another build and see if this is still an issue.

**Comment by simon-mo** [comment]
Pending on #15925

**Pull request merged**